doc/coq.rst

```diff
@@ -0,0 +1,392 @@
+.. _coq:
+
+***
+Coq
+***
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Introduction
+------------
+
+Dune can build Coq theories and plugins with additional support for extraction
+and ``.mlg`` file preprocessing.
+
+A *Coq theory* is a collection of ``.v`` files all prefixed by a common Coq
+module prefix, following the directory hierarchy for module names.
+
+A *Coq plugin* is an OCaml :ref:`library` that is intended to be dynamically
+loaded by Coq at runtime when compiling a Coq theory, which is typically linked
+with the OCaml Coq API.
```

Comment:
```suggestion
A *Coq plugin* is an OCaml :ref:`library` that Coq can load dynamically
at runtime.  Plugins are typically linked with the Coq OCaml API.
```

Not just "compiling", which doesn't include the IDEs.

Reply:
Better yes!